core/src/main/java/io/lionweb/lioncore/java/model/impl/AnnotatedNode.java

```diff
@@ -0,0 +1,77 @@
+package io.lionweb.lioncore.java.model.impl;
+
+import io.lionweb.lioncore.java.language.Annotation;
+import io.lionweb.lioncore.java.language.Classifier;
+import io.lionweb.lioncore.java.model.AnnotationInstance;
+import io.lionweb.lioncore.java.model.ClassifierInstance;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import javax.annotation.Nonnull;
+
+public abstract class AnnotatedNode<T extends Classifier<T>> implements ClassifierInstance<T> {
```

Comment:
What is an AnnotatedNode?

Reply:
If I undestand correctly, this represents something that _can be annotated_, not something that is necessarily _annotated_. Also, I am confused by the term node.

I think we have:
- members of the tree (without annotation instances). I call these nodes
- annotation instances
- members of the tree + annotations. These are classifierInstances

I understand that an annotation can be attached both to nodes and annotation instances.

So I would could this class `AnnotationInstanceTarget` or `AnnotableClassifierInstance` or just `ClassifierInstanceBase` or `AbstractClassifierInstance`